src/Symfony/Bridge/PhpUnit/bin/simple-phpunit

```diff
@@ -57,10 +57,11 @@ foreach ($defaultEnvs as $envName => $envValue) {
     }
 }
 
-$COMPOSER = file_exists($COMPOSER = $oldPwd.'/composer.phar')
+$COMPOSER =
+    file_exists($COMPOSER = $oldPwd.'/composer.phar')
     || ($COMPOSER = rtrim('\\' === DIRECTORY_SEPARATOR ? preg_replace('/[\r\n].*/', '', `where.exe composer.phar`) : `which composer.phar 2> /dev/null`))
     || ($COMPOSER = rtrim('\\' === DIRECTORY_SEPARATOR ? preg_replace('/[\r\n].*/', '', `where.exe composer`) : `which composer 2> /dev/null`))
-    ? $PHP.' '.escapeshellarg($COMPOSER)
+    ? (!'\\' === DIRECTORY_SEPARATOR && (fread($fp = fopen($COMPOSER, 'rb'), 2) === '#!' && fclose($fp)) ? '' : $PHP).' '.escapeshellarg($COMPOSER) // checking that detected COMPOSER has shebang at the start, if it's found do not append php-executable
```

Comment:
I think this would work better:
`? (file_get_contents($COMPOSER, false, null, 0, 18) === '#!/usr/bin/env php' ? $PHP : '').' '.escapeshellarg($COMPOSER) // detect shell wrappers by looking at the shebang`
this works on linux and windows, and detects php exactly. This shebang is the one that composer uses since always

Reply:
Shame to me 😟 , I didn't know that `file_get_contents` could read file with offset.

Reply:
no worries, I copy/pasted from SO :)

Reply:
done